Fix a few special cases of projects with command line arguments

trylek opened the pull request:
This tackles the few special cases where command-line arguments are either ignored or used to run the test with a specific set of arguments; in the latter case I reworked the entry point to accept the parameters in its signature. This change doesn't address those cases where we run the same csproj test multiple times with different command-line combinations; that will require further design discussion with the GC team that owns most of the tests in this category.

Thanks

Tomas

Contributes to: https://github.com/dotnet/runtime/issues/54512

trylek committed:
Remove CLRTestExecutionArguments from OOMException01

Test entrypoint is parameterless and so it ignores the argument

trylek committed:
The test entrypoint for 347011 is parameterless so the property is ignored

trylek committed:
Remove project-driven command line arguments from the polyrec test

The test just passes a pair of integers that are used as internal
parameters for the test. Refactor the test so that the entrypoint
accepts the parameters as arguments and call it once from Main
the way we used to do based on the CLRTestExecutionArguments.

Thanks

Tomas

trylek commented:
/azp run runtime-coreclr outerloop

BruceForstall commented on src/tests/Loader/regressions/polyrec/polyrec.cs:
```diff
@@ -63,24 +63,24 @@ public static void Start2()
   }
 
-  public static int Main(String[] args)
+  public static int Test(int threads, int insts)
```
In a case like this, would it make sense to keep the `Main` that parses args and passes them to `Test`, which you've created, but also add a test function that can be marked with `[Fact]` to call it with the precise arguments the test case is using, e.g.,

```
[Fact]
public static int Test1()
{
   return Test(4, 50);
}
```

the idea being, if you want to run the standalone test manually with arbitrary arguments, you still can, but the test harness calling the `[Fact]` function will get the specific arguments.

Maybe for this one it doesn't matter, but I'm thinking maybe some of the JIT benchmarks might be like this.

(btw, unrelated, note that when you get to `src\tests\JIT\Performance\CodeQuality`, those are "as close as we can make them" to the copies in the dotnet/performance repo at `src\benchmarks\micro\runtime`)

trylek committed:
Fix the polyrec.cs test based on PR feedback

Based on PR feedback and Jeremy's check-in of the Roslyn
generator test logic I can now fully update the test so that it
no longer contains the Main method and it just implements
the [Fact]-marked test entrypoint.

Thanks

Tomas

trylek committed:
Remove the exe test output type from the polyrec.csproj project

trylek commented:
Merging in with two unrelated bugs in library tests that are unaffected by this change.
